src/esp/gfx/replay/Recorder.cpp

```diff
@@ -216,6 +237,33 @@ void Recorder::updateInstanceStates() {
       instanceRecord.recentState = state;
     }
   }
+
+  // Update rig states
+  for (const auto& rigItr : rigNodes_) {
```

Comment:
How about don't write an update if the rig state hasn't changed? Similar to above logic.